src/main/webapp/WEB-INF/web.xml

```diff
@@ -5,14 +5,54 @@
          version="3.1">
   <display-name>Phoneshop servlet api training</display-name>
 
+  <context-param>
+    <param-name>insertDemoData</param-name>
+    <param-value>true</param-value>
+  </context-param>
+
+  <listener>
+    <listener-class>com.es.phoneshop.web.ProductDemodataServletContextListener</listener-class>
+  </listener>
+
   <servlet>
     <servlet-name>productList</servlet-name>
     <servlet-class>com.es.phoneshop.web.ProductListPageServlet</servlet-class>
   </servlet>
-  
+
   <servlet-mapping>
     <servlet-name>productList</servlet-name>
     <url-pattern>/products</url-pattern>
   </servlet-mapping>
 
+  <servlet>
+    <servlet-name>productInfo</servlet-name>
+    <servlet-class>com.es.phoneshop.web.ProductDetailsPageServlet</servlet-class>
+  </servlet>
+  
+  <servlet-mapping>
+    <servlet-name>productInfo</servlet-name>
+    <url-pattern>/products/info/*</url-pattern>
+  </servlet-mapping>
+
+  <servlet>
+    <servlet-name>productHistory</servlet-name>
+    <servlet-class>com.es.phoneshop.web.ProductPriceHistoryServlet</servlet-class>
+  </servlet>
+
+  <servlet-mapping>
+    <servlet-name>productHistory</servlet-name>
+    <url-pattern>/products/history/*</url-pattern>
+  </servlet-mapping>
+
+  <error-page>
+  	<location>/WEB-INF/pages/productNotFoundError.jsp</location>
+  	<exception-type>com.es.phoneshop.exception.ProductNotFoundException</exception-type>
+  </error-page>
+<!-- 
+  <error-page>
```

Comment:
You have to create error pages for 404 nad 500 errors. Why have you commented it?

Reply:
I forgot to uncomment them.
Default exception page is useful when something goes wrong.